### DISCUSSION TITLE
Help with performing Flux balance analysis

### DISCUSSION THREAD

**Discussion by abshah1** [opencobra/cobrapy#1079] [General]
Hi,

I'm new to FBA analysis with escher and cobrapy. I have created a metabolic map (around 28 reactions) on escher using an exisiting cobra model (called iJN678.json which contains 863 reactions). I saved the map as a json file, (called iJN678test.json). However, the model is not updated with the 28 reactions only ( i.e. 863 reactions are still present in the model). As a result the FBA analysis is perfromed for 863 reactions (iJN678.json) instead of the 28 reactions specified by the metabolic map. My issue is how do I create a model with just the 28 reactions specified in my metabolic map to perform FBA analysis. I have attached screenshots of the escher map and the code to check for the number of reactions in my metabolic network (new model iJN678test.json)

![Screenshot 2021-04-13 133148](https://user-images.githubusercontent.com/79981819/114552565-a4c17c00-9c5c-11eb-87fd-580a00f6c8d3.jpg)
![Screenshot 2021-04-13 132954](https://user-images.githubusercontent.com/79981819/114552567-a55a1280-9c5c-11eb-8577-ef20aa3f774f.jpg)

**Comment by Midnighter** [comment]
Hi @abshah1,

Thanks for using the discussion forum, I think it will be a helpful feature in the long run 🙂 Regarding your question: I doubt that your model would be in a functional state if you restrict it to the 28 reactions. My approach would be to generate a flux solution using the whole model and then map the fluxes that exist on your map, for example,

```py
import escher
import cobra.flux_analysis as cflux
from cobra.io import load_model

model = load_model("iJN678")
solution = cflux.pfba(model)

builder = escher.Builder(
    model=model,
    map_json="iJN678test.json",
    reaction_data=solution.fluxes,
)
builder
```

**Reply by abshah1** [reply]
Hi @Midnighter 

Thank you for your response, it was a lot of help! I have a couple more questions. My metabolic map has grown, I've added the MEP pathway, with the goal of maximising flux through geranyl diphosphate. I tried running FBA on EscherFBA but the flux is relatively low in the MEP pathway compared to other fluxes in the network. In literature it is expected as the first reaction is the limiting reaction, but I am trying to increase flux by using optknock but I cannot get it working in Escher, is there a way to increase flux, or should I try running it on python instead of EscherFBA? I have attached a picture to show fluxes in Escher
![Screenshot 2021-04-29 121749](https://user-images.githubusercontent.com/79981819/116543061-4dbad880-a8e5-11eb-945a-452ab84c47e6.jpg)

**Reply by Midnighter** [reply]
As far as I'm aware, EscherFBA can only perform FBA. So you need to use a different tool for OptKnock indeed. Since you like to work with the map interface, I suggest you take a look at https://caffeine.dd-decaf.eu/ where you can perform computational strain engineering tasks.
